drivers/sensor/stm32_temp/stm32_temp.c

```diff
@@ -24,10 +33,19 @@ struct stm32_temp_data {
 };
 
 struct stm32_temp_config {
+	int tsv_mv;
+#if HAS_CALIBRATION == 0
 	int avgslope;
 	int v25_mv;
-	int tsv_mv;
 	bool is_ntc;
+#else
+	uint16_t *cal1_addr;
+	uint16_t *cal2_addr;
+	int cal1_temp;
+	int cal2_temp;
+	int cal_vrefanalog;
+	int cal_offset;
+#endif
```

Comment:
Nit: instead of negative logic, let's use `#if HAS_CALIBRATION` and keep the same order as you used below.

Reply:
Is indeed more consistent. Done.